vessim/sil.py

```diff
@@ -25,76 +26,96 @@
 from vessim._util import DatetimeLike
 
 
-def _iterate_queue(q: multiprocessing.Queue, timeout: Optional[float] = None) -> Iterable[Any]:
-    blocking = timeout is not None
-    while True:
-        try:
-            yield q.get(blocking, timeout)
-        except QueueEmpty:
-            break
+class Broker:
+    def __init__(self, data_pipe_out: Connection, events_pipe_in: Connection):
+        self._data_pipe_out = data_pipe_out
+        self._events_pipe_in = events_pipe_in
+        self._microgrid_ts: list[tuple[DatetimeLike, Microgrid]] = []
+        self._actor_infos_ts: list[tuple[DatetimeLike, dict]] = []
```

Comment:
If the signature of `get_actor_ts` is updated, it should be updated here, too of course.
```suggestion
        self._actor_infos_ts: list[tuple[DatetimeLike, dict[str, dict]]] = []
```

Although this becomes a bit verbose... One could introduce a type alias like this, if you wanted to:
```
T = TypeVar("T")
TimeSeries = list[tuple[DatetimeLike, T]]
```
And then write
```
        self._actor_infos_ts: TimeSeries[dict[str, dict]] = []
```

Reply:
Ah, on the note of types... since vessim apparently tries to support python 3.8 (and I use 3.8 myself)... I should have not used the type signatures from 3.9 in my example, because things like `tuple[int]` do not work below 3.9.
Instead it needs to be like this:
```suggestion
        self._actor_infos_ts: List[Tuple[DatetimeLike, Dict[str, dict]]] = []
```
With the corresponding imports from `typing`.

And of course, this would need to be changed all over the newly added code. But I would also really appreciate if I could continue to use 3.8 since this is the only version available to ubuntu 20.04 :)

Reply:
Ah yes, my bad. Always forget that ^^ No, we will continue to support 3.8 for now. The use of `from __future__ import annotations` allows us to use 3.8+ type annotations in function headers, but in function bodies :S

Reply:
> The use of `from __future__ import annotations` allows us to use 3.8+ type annotations in function headers

Now that's something I didn't know :)